src/panels/developer-tools/template/developer-tools-template.ts

```diff
@@ -378,11 +382,36 @@ class HaPanelDevTemplate extends LitElement {
     localStorage["panel-dev-template-template"] = this._template;
   }
 
-  private _restoreDemo() {
+  private async _restoreDemo() {
+    if (
+      !(await showConfirmationDialog(this, {
+        text: this.hass.localize(
+          "ui.panel.developer-tools.tabs.templates.confirm_reset"
+        ),
+        warning: true,
+      }))
+    ) {
+      return;
+    }
     this._template = DEMO_TEMPLATE;
     this._subscribeTemplate();
     delete localStorage["panel-dev-template-template"];
   }
+
+  private async _clear() {
+    if (
+      !(await showConfirmationDialog(this, {
+        text: this.hass.localize(
+          "ui.panel.developer-tools.tabs.templates.confirm_clear"
+        ),
+        warning: true,
+      }))
+    ) {
+      return;
+    }
+    this._template = "";
+    this._subscribeTemplate();
```

Comment:
```suggestion
    this._unsubscribeTemplate();
```

Reply:
If we don't have a template, it makes no sense to subscribe to it.

Reply:
If we omit that call, then the template preview panel remains in an incorrect old state. If however I leave that call in, then it updates/looks the same it would if I manually clear the template input (not via the new button).

Reply:
That can be fixed with css right?

Reply:
> That can be fixed with css right?

Sure it can, but we then still end up with different behaviour between manually clearing the template input and using the clear button. But not a hill I am willing to die on, as we are in dev tools.

Reply:
I'm also fine with setting `_templateResult` to:

```
{
    "result": "",
    "listeners": {
        "all": false,
        "entities": [],
        "domains": [],
        "time": false
    }
}
```

Reply:
I like that approach as we then stay visually consistent. Will push a change.